File: backend/models/product.py
from pydantic import BaseModel, Field
from typing import Optional
from uuid import UUID
from datetime import datetime

class ProductBase(BaseModel):
    product_name: Optional[str] = None
    short_title: Optional[str] = None
    category: Optional[str] = None
    subcategory: Optional[str] = None
    description: Optional[str] = None
    materials: Optional[list[str]] = Field(default_factory=list)
    colors: Optional[list[str]] = Field(default_factory=list)
    pattern: Optional[str] = None
    style: Optional[str] = None
    shape: Optional[str] = None
    finish: Optional[str] = None
    target_audience: Optional[list[str]] = Field(default_factory=list)
    features: Optional[list[str]] = Field(default_factory=list)
    benefits: Optional[list[str]] = Field(default_factory=list)
    use_cases: Optional[list[str]] = Field(default_factory=list)
    selling_points: Optional[list[str]] = Field(default_factory=list)
    seo_keywords: Optional[list[str]] = Field(default_factory=list)
    tags: Optional[list[str]] = Field(default_factory=list)
    price_min: Optional[float] = None
    price_max: Optional[float] = None
    confidence: Optional[float] = None

class ProductCreate(ProductBase):
    image_url: str

class ProductUpdate(ProductBase):
    pass

class ProductResponse(ProductBase):
    id: UUID
    user_id: UUID
    image_url: str
    created_at: datetime
    updated_at: datetime

class PaginatedProducts(BaseModel):
    data: list[ProductResponse]
    total: int
    page: int
    limit: int

class AnalysisResponse(ProductBase):
    pass

File: frontend/src/types/product.ts
export interface Product {
  id: string;
  user_id: string;
  image_url: string;
  product_name: string | null;
  short_title: string | null;
  category: string | null;
  subcategory: string | null;
  description: string | null;
  materials: string[];
  colors: string[];
  pattern: string | null;
  style: string | null;
  shape: string | null;
  finish: string | null;
  target_audience: string[];
  features: string[];
  benefits: string[];
  use_cases: string[];
  selling_points: string[];
  seo_keywords: string[];
  tags: string[];
  price_min: number | null;
  price_max: number | null;
  confidence: number | null;
  created_at: string;
  updated_at: string;
}

export interface ProductUpdate {
  product_name?: string | null;
  short_title?: string | null;
  category?: string | null;
  subcategory?: string | null;
  description?: string | null;
  materials?: string[];
  colors?: string[];
  pattern?: string | null;
  style?: string | null;
  shape?: string | null;
  finish?: string | null;
  target_audience?: string[];
  features?: string[];
  benefits?: string[];
  use_cases?: string[];
  selling_points?: string[];
  seo_keywords?: string[];
  tags?: string[];
  price_min?: number | null;
  price_max?: number | null;
  confidence?: number | null;
}

export interface UploadResponse {
  image_url: string;
  filename: string;
}

export interface BulkUploadResponse {
  image_urls: string[];
}

export interface AnalyzeRequest {
  image_url: string;
}

export interface BulkAnalyzeRequest {
  image_urls: string[];
}

export interface PaginatedResponse<T> {
  data: T[];
  total: number;
  page: number;
  limit: number;
}

export type FileStatus = 'idle' | 'uploading' | 'analyzing' | 'complete' | 'error';

export interface UploadFileItem {
  id: string;
  file: File;
  preview: string;
  status: FileStatus;
  progress: number;
  imageUrl?: string;
  product?: Product;
  error?: string;
}
